src/AppBundle/Form/UserType.php

```diff
@@ -11,6 +12,20 @@
 
 class UserType extends AbstractType
 {
+    /**
+     * @var array
+     */
+    private $roles = [];
+
+    public function __construct(array $securityRoleHierarchyRoles, array $translations)
+    {
+        if ([] !== $missingTranslations = array_diff(array_keys($securityRoleHierarchyRoles), array_keys($translations))) {
+            throw new \InvalidArgumentException("Missing translations: " . print_r($missingTranslations, true));
+        }
+
+        $this->roles = array_flip($translations);
```

Comment:
Flip your array definition and work with `array_values` instead of `array_keys` to improve readability ;)

Reply:
Do you mean doing something like that:
```php
        if ([] !== $missingTranslations = array_diff(array_flip($securityRoleHierarchyRoles), array_flip($translations))) {
            throw new \InvalidArgumentException("Missing translations: " . print_r($missingTranslations, true));
        }
    
        $this->roles = $translations;
```

If so, then I get **Warning: array_flip(): Can only flip STRING and INTEGER values!**.

By the way, I refactored UserType to be more readable even while working with keys [with this commit](https://github.com/nicordev/formation-oc-php-projet8-todoco/pull/11/commits/b40ca33a79291b7fb4d65e0fbd0b6092933ffe92).